core/src/main/scala/org/bitcoins/core/crypto/Sign.scala

```diff
@@ -26,27 +27,84 @@ trait Sign {
   def signFuture(bytes: ByteVector): Future[ECDigitalSignature] =
     signFunction(bytes)
 
+  def signWithEntropyFunction: (
+      ByteVector,
+      ByteVector) => Future[ECDigitalSignature]
+
+  def signWithEntropyFuture(
+      bytes: ByteVector,
+      entropy: ByteVector): Future[ECDigitalSignature] =
+    signWithEntropyFunction(bytes, entropy)
+
+  private def signLowRFuture(bytes: ByteVector, startAt: UInt32)(
```

Comment:
`@tailrec`

Reply:
actually it isn't tail recursive sadly due to the fact that the call is made within a `map`

Reply:
:(